bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVaneerSettingsFactory.java

```diff
@@ -0,0 +1,298 @@
+   * @throws IOException if any.
+   */
+  public static BigtableDataSettings createBigtableDataSettings(@Nonnull final BigtableOptions options) throws IOException {
+    checkState(options.getProjectId() != null, "Project ID is required");
+    checkState(options.getInstanceId() != null, "Instance ID is required");
+    checkState(options.getRetryOptions().enableRetries(), "Disabling retries is not currently supported.");
+
+    final BigtableDataSettings.Builder builder = BigtableDataSettings.newBuilder();
+    final String endpoint = options.getDataHost() + ":" + options.getPort();
+
+    builder.setProjectId(options.getProjectId());
+    builder.setInstanceId(options.getInstanceId());
+    builder.setAppProfileId(options.getAppProfileId());
+
+    builder.setEndpoint(endpoint);
+
+    builder.setCredentialsProvider(buildCredentialProvider(options.getCredentialOptions()));
+
+    buildBulkMutationsSettings(builder, options);
+
+    buildCheckAndMutateRowSettings(builder, options.getCallOptionsConfig().getShortRpcTimeoutMs());
+
+    buildReadModifyWriteSettings(builder, options.getCallOptionsConfig().getShortRpcTimeoutMs());
+
+    buildReadRowsSettings(builder, options);
+
+    buildMutateRowSettings(builder, options);
+
+    buildSampleRowKeysSettings(builder, options);
+
+    String userAgent = BigtableVersionInfo.CORE_USER_AGENT + "," + options.getUserAgent();
+    HeaderProvider headers = FixedHeaderProvider.create(GrpcUtil.USER_AGENT_KEY.name(), userAgent);
+
+    builder.setTransportChannelProvider(
+        InstantiatingGrpcChannelProvider.newBuilder()
+            .setHeaderProvider(headers)
+            .setEndpoint(endpoint)
+            .setPoolSize(options.getChannelCount())
+            .build());
+
+    return builder.build();
+  }
+
+  /**
+   * To create an instance of {@link BigtableTableAdminSettings} from {@link BigtableOptions}.
+   *
+   * @param options a {@link BigtableOptions} object.
+   * @return a {@link BigtableTableAdminSettings} object.
+   * @throws IOException if any.
+   */
+  public static BigtableTableAdminSettings createTableAdminSettings(@Nonnull final BigtableOptions options)
+      throws IOException {
+    checkState(options.getProjectId() != null, "Project ID is required");
+    checkState(options.getInstanceId() != null, "Instance ID is required");
+
+    final BigtableTableAdminSettings.Builder adminBuilder = BigtableTableAdminSettings.newBuilder();
+
+    adminBuilder.setProjectId(options.getProjectId());
+    adminBuilder.setInstanceId(options.getInstanceId());
+
+    final String endpoint = options.getAdminHost() + ":" + options.getPort();
+    adminBuilder.stubSettings().setEndpoint(endpoint);
+
+    //Overriding default credential with BigtableOptions's credential.
+    adminBuilder.stubSettings()
+        .setCredentialsProvider(buildCredentialProvider(options.getCredentialOptions()));
+
+    String userAgent = BigtableVersionInfo.CORE_USER_AGENT + "," + options.getUserAgent();
```

Comment:
Please add something here that would allow us to disambiguate when the hbase client is using the gcj adapter

Reply:
Does ["VENEER_ADAPTER"](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/1973/files#diff-6a492166d0005302ce3c542cdcf00ad5R54) sound good?